pontoon/administration/tests/test_sync_projects.py

```diff
@@ -0,0 +1,655 @@
+            self.changeset, self.db_project, key, db_entity, vcs_entity
+        )
+
+    def test_handle_entity_none(self):
+        """
+        If both the db_entity and vcs_entity are None, raise a
+        CommandError, as that should never happen.
+        """
+        with assert_raises(CommandError):
+            self.call_handle_entity('key', None, None)
+
+    def test_handle_entity_obsolete(self):
+        """If VCS is missing the entity in question, obsolete it."""
+        self.create_db_entities_translations()
+        self.changeset.obsolete_db_entity = Mock()
+        self.call_handle_entity('key', self.main_db_entity, None)
+        self.changeset.obsolete_db_entity.assert_called_with(self.main_db_entity)
+
+    def test_handle_entity_create(self):
+        """If the DB is missing an entity in VCS, create it."""
+        self.create_db_entities_translations()
+        self.changeset.create_db_entity = Mock()
+        self.call_handle_entity('key', None, self.main_vcs_entity)
+        self.changeset.create_db_entity.assert_called_with(self.main_vcs_entity)
+
+    def test_handle_entity_no_translation(self):
+        """If no translation exists for a specific locale, skip it."""
+        self.create_db_entities_translations()
+        self.changeset.update_vcs_entity = Mock()
+        self.changeset.update_db_entity = Mock()
+        self.main_vcs_entity.has_translation_for = Mock(return_value=False)
+
+        self.call_handle_entity('key', self.main_db_entity, self.main_vcs_entity)
+        assert_false(self.changeset.update_vcs_entity.called)
+        assert_false(self.changeset.update_db_entity.called)
+
+    def test_handle_entity_db_changed(self):
+        """
+        If the DB entity has changed since the last sync, update the
+        VCS.
+        """
+        self.create_db_entities_translations()
+        self.changeset.update_vcs_entity = Mock()
+        with patch.object(Entity, 'has_changed', return_value=True):
+            self.call_handle_entity('key', self.main_db_entity, self.main_vcs_entity)
+
+        self.changeset.update_vcs_entity.assert_called_with(
+            self.translated_locale.code, self.main_db_entity, self.main_vcs_entity
+        )
+
+    def test_handle_entity_vcs_changed(self):
+        """
+        If the DB entity has not changed since the last sync, update the DB with
+        the latest changes from VCS.
+        """
+        self.create_db_entities_translations()
+        self.changeset.update_db_entity = Mock()
+        with patch.object(Entity, 'has_changed', return_value=False):
+            self.call_handle_entity('key', self.main_db_entity, self.main_vcs_entity)
+
+        self.changeset.update_db_entity.assert_called_with(
+            self.translated_locale.code, self.main_db_entity, self.main_vcs_entity
+        )
+
+    def test_update_resources(self):
+        # Check for self.main_db_resource to be updated and
+        # self.other_db_resource to be created.
```

Comment:
I didn't make it a docstring because the comment isn't really explaining the purpose of the test, it's explaining the specifics of how this is testing things. Sometimes the test docstrings are used in place of the test name, and I feel like this comment would be a bad name for this test.